beacon/light/api/light_api.go

```diff
@@ -499,9 +507,48 @@ func (api *BeaconLightApi) StartHeadListener(listener HeadEventListener) func()
 			}
 		}
 	}()
+
 	return func() {
-		close(closeCh)
-		<-closedCh
-		<-stoppedCh
+		closeCtx()
+		wg.Wait()
+	}
+}
+
+// startEventStream establishes an event stream. This will keep retrying until the stream has been
+// established. It can only return nil when the context is canceled.
+func (api *BeaconLightApi) startEventStream(ctx context.Context, listener *HeadEventListener) *eventsource.Stream {
+	for initial := true; ; initial = false {
+		if !initial {
+			if ctxSleep(ctx, 5*time.Second) {
+				return nil
+			}
+		}
```

Comment:
This is a bit clunky and un-intuitive. I have a less clunky but perhaps also un-intuitive suggestion:
 
```suggestion
	for ;; shouldRetry( ctx ){
```
And then you have 
```golang
// shouldRetry waits up to 5 seconds and returns 'true', or 'false' if the context becomes done
func shouldRetry(ctx context.Context, timeout time.Duration) (bool) {
	timer := time.NewTimer(5 * time.Second)
	defer timer.Stop()
	select {
	case <-timer.C:
		return true // retry
	case <-ctx.Done():
		return false // don't retry, we're done
	}
}
```